pkg/controller/networkaddonsconfig/networkaddonsconfig_controller_test.go

```diff
@@ -0,0 +1,123 @@
+package networkaddonsconfig
+
+import (
+	"fmt"
+
+	"github.com/kubevirt/cluster-network-addons-operator/pkg/names"
+	"github.com/kubevirt/cluster-network-addons-operator/pkg/render"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type checkUnit struct {
+	key           string
+	shouldExist   bool
+	expectedValue string
+}
+
+var _ = Describe("Networkaddonsconfig", func() {
+	Context("When CR is not labeled", func() {
+		It("Should find only default labels", func() {
+			renderData := render.MakeRenderData()
+			objs, err := render.RenderTemplate("testdata/data.yaml", &renderData)
+			Expect(err).NotTo(HaveOccurred())
+
+			crLables := map[string]string{}
+			err = updateObjectsLabels(crLables, objs)
+			Expect(err).NotTo(HaveOccurred())
+
+			appLabelKeys := []checkUnit{
+				{
+					key:           names.COMPONENT_LABEL_KEY,
+					shouldExist:   true,
+					expectedValue: names.COMPONENT_LABEL_DEFAULT_VALUE,
+				},
+				{
+					key:           names.MANAGED_BY_LABEL_KEY,
+					shouldExist:   true,
+					expectedValue: names.MANAGED_BY_LABEL_DEFAULT_VALUE,
+				},
+				{
+					key:           names.PART_OF_LABEL_KEY,
+					shouldExist:   false,
+					expectedValue: "Invalid",
+				},
+				{
+					key:           names.VERSION_LABEL_KEY,
+					shouldExist:   false,
+					expectedValue: "Invalid",
+				},
+			}
+
+			labels := objs[0].GetLabels()
+			err = checkLabels(labels, appLabelKeys)
+			Expect(err).NotTo(HaveOccurred())
+
+			templateLabels, _, _ := unstructured.NestedStringMap(objs[0].Object, "spec", "template", "metadata", "labels")
+			err = checkLabels(templateLabels, appLabelKeys)
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+	Context("When CR is labeled", func() {
+		It("Should find all labels, overrided by CR labels", func() {
+			renderData := render.MakeRenderData()
+			objs, err := render.RenderTemplate("testdata/data.yaml", &renderData)
+			Expect(err).NotTo(HaveOccurred())
+
+			const expectedManagedByLabel = "managed_by_unit_tests"
```

Comment:
the whole part configuring the crLables should also be in `BeforeEach()`

Reply:
But its different per each `It`